internal/proxyservice/http/aws/aws.go

```diff
@@ -0,0 +1,191 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	gohttp "net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+)
+
+// From https://github.com/aws/aws-sdk-go/blob/master/aws/signer/v4/v4.go#L77
+const (
+	timeFormat = "20060102T150405Z"
+)
+
+// newAmzDate parses a date string using the AWS signer time format
+func newAmzDate(amzDateStr string) (time.Time, error) {
+	if amzDateStr == "" {
+		return time.Time{}, fmt.Errorf("missing required header: %s", "X-Amz-Date")
+	}
+
+	return time.Parse(timeFormat, amzDateStr)
+}
+
+// requestMetadataFromAuthz parses an authorization header string and create a
+// requestMetadata instance populated with the associated region and service
+// name
+func requestMetadataFromAuthz(authorizationStr string) (*requestMetadata, error) {
```

Comment:
This is certainly not a change you need to make, since the current code is clear, but more of a "just pointing out a design pattern you can use in the future"....

That said, what's happening here is analogous to the `NetworkAddress` type that emerged during our previous work this week.  It seems that the "authorization header" is telling us that it wants to be a type.  That is, we are already _thinking of it_ as its own type, and in particular as an object which knows its own region and serviceName.

Reply:
I see what you're getting at and suspect it might require more thought to design, so I'd opt to defer.